src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3735,6 +3772,48 @@ class CgroupUtils(object):
                 raise
         return self.assigned_resources is not None
 
+    def add_jobid_to_cgroup_jobs(self, jobid):
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'Adding jobid %s to cgroup_jobs' % jobid)
+        with open(self.cgroup_jobs_file, 'a+') as f:
```

Comment:
What if the jobid is already in the list? Do you still want to append it?
What if the open fails? Should you add a try/except block? If you do, don't forget to add an except for KeyboardInterrupt.

Reply:
Yes.  When the list is read, it is read into a set so duplicates will be removed. 
I will add a try/except.